### PR TITLE
doc(components): Added component index in the readme

### PR DESCRIPTION
**Pull request by Ark-kun**
Also added a script to [re-]generate the index.

### CODE REVIEW

File: components/README.md [hunk 1]
```diff
@@ -33,3 +33,155 @@ Note the naming convention for client code and runtime code&mdash;for a task nam
 
 See how to [use the Kubeflow Pipelines SDK](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/)
 and [build your own components](https://www.kubeflow.org/docs/pipelines/sdk/build-component/).
+
+## Index of components
+ / [_converters](https://github.com/kubeflow/pipelines/tree/b63dd3f53237f74ae7f965441fd97698fafba7c9/components/_converters) / [ApacheParquet](https://github.com/kubeflow/pipelines/tree/b63dd3f53237f74ae7f965441fd97698fafba7c9/components/_converters/ApacheParquet) / [from_ApacheArrowFeather](https://github.com/kubeflow/pipelines/tree/b63dd3f53237f74ae7f965441fd97698fafba7c9/components/_converters/ApacheParquet/from_ApacheArrowFeather) / [Convert apache arrow feather to apache parquet](https://raw.githubusercontent.com/kubeflow/pipelines/b63dd3f53237f74ae7f965441fd97698fafba7c9/components/_converters/ApacheParquet/from_ApacheArrowFeather/component.yaml)
```
Comment (numerology) [inline components/README.md:38]: nit: We might need a new line after each entry. Currently it looks very busy. WDYT?

File: components/generate_component_index.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+with open('README.md', 'r') as f:
+    readme_lines = f.readlines()
+
+with open('README.md', 'w') as f:
+    for line in readme_lines:
+        f.write(line)
+        if 'Index of components' in line:
+            break
+    for line in component_list_lines:
+        f.write(line + '\n')
```
Comment (numerology) [inline components/generate_component_index.py:58]: I guess we need double '\n's here due to how markdown is rendered.

**Review by numerology — COMMENTED** [review]
Thanks!

File: components/README.md (hunk 1 above)
Comment (Ark-kun) [reply components/README.md:38]: Thank you for catching this. The markdown preview in VSCode is incorrect =(
Fixed.

File: components/generate_component_index.py (hunk 2 above)
Comment (Ark-kun) [reply components/generate_component_index.py]: Thank you.
Fixed.

**Pull request merged**